service/controller/resource/prometheusremotewrite/resource_test.go

```diff
@@ -0,0 +1,396 @@
+			}
+		})
+	}
+}
+
+func TestRemovePrometheusRemoteWrite(t *testing.T) {
+
+	rw := remoteWrite(name, namespace, clusterSelector)
+	prom := prometheus()
+	expectedPrometheus := expectedPrometheusEmptyRemoteWrite(rw, prom)
+	expectedRemoved := expectedRemoveRemoteWrite(*expectedPrometheus.p.DeepCopy(), 0)
+
+	rwNotFound := remoteWrite("remotewrite-notfound", namespace, clusterSelector)
+	expectedNotFound := expectedRemoveRemoteWriteNotExists(*expectedPrometheus.p.DeepCopy())
+
+	type args struct {
+		rw pmov1alpha1.RemoteWrite
+		p  promv1.Prometheus
+	}
+
+	type want struct {
+		p  *promv1.Prometheus
+		ok bool
+	}
+
+	cases := map[string]struct {
+		reason string
+		args   args
+		want   want
+	}{
+		"RemoteWriteRemoved": {
+			reason: "RemoteWrite removed from Prometheus",
+			args: args{
+				rw: rw,
+				p:  *expectedPrometheus.p.DeepCopy(),
+			},
+			want: want(expectedRemoved),
+		},
+		"RemoteWriteNotFound": {
+			reason: "RemoteWrite not found in Prometheus",
+			args: args{
+				rw: rwNotFound,
+				p:  *expectedPrometheus.p.DeepCopy(),
+			},
+			want: want(expectedNotFound),
+		},
+	}
+
+	for n, tc := range cases {
+		t.Run(n, func(t *testing.T) {
+			got, ok := removePrometheusRemoteWrite(tc.args.rw, tc.args.p)
+			if tc.want.ok != ok {
+				t.Errorf("\n%s\nExpand(...): -want, +got:\n%v", tc.reason, ok)
+			}
+			if diff := cmp.Diff(tc.want.p, got); diff != "" {
+				t.Errorf("\n%s\nExpand(...): -want, +got:\n%s", tc.reason, diff)
+			}
+		})
+	}
+}
+
+func remoteWrite(name, namespace, clusterSelector string) pmov1alpha1.RemoteWrite {
+
+	return pmov1alpha1.RemoteWrite{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: pmov1alpha1.RemoteWriteSpec{
+			ClusterSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"cluster": clusterSelector},
+			},
+			RemoteWrite: promv1.RemoteWriteSpec{
+				URL:  "https://my-fancy-url",
+				Name: "test",
+				BasicAuth: &promv1.BasicAuth{
+					Username: corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: name,
+						},
+						Key: "username",
+					},
+					Password: corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: name,
+						},
+						Key: "password",
+					},
+				},
+			},
+		},
+	}
+}
+
+func prometheus() promv1.Prometheus {
+	var replicas int32 = 1
+	return promv1.Prometheus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "some-prometheus",
+			Namespace: "default",
+		},
+		Spec: promv1.PrometheusSpec{
+			Version:  "v1.35.0",
+			Replicas: &replicas,
+		},
+	}
+}
+
+func expectedPrometheusEmptyRemoteWrite(rw pmov1alpha1.RemoteWrite, prom promv1.Prometheus) struct {
+	p  *promv1.Prometheus
+	ok bool
+} {
+	rw.Spec.RemoteWrite.Name = rw.GetName()
+
+	r := NewResource()
+	if len(r.HTTPSProxy) > 0 {
+		rw.Spec.RemoteWrite.ProxyURL = r.HTTPSProxy
+	} else if len(r.HTTPProxy) > 0 {
+		rw.Spec.RemoteWrite.ProxyURL = r.HTTPProxy
+	}
```

Comment:
Why do we have this everywhere?

I would rather have a different test for the http proxy, instead of having this piece of code duplicated in every test.

Reply:
because `ensurePrometheusRemoteWrite()` have it, so it returns the proxy in the remotewrite config.
If don't use them, tests will fail.

Reply:
I added `TestEnsurePrometheusRemoteWriteWithPoxy()` and I added an argument `proxy bool` to  `expectedPrometheusAppend()`.